src/config/certificates.rs

```diff
@@ -91,3 +129,80 @@ fn generate_key_pair() -> Result<(X509, PKey<Private>), ErrorStack> {
 
     Ok((cert, privkey))
 }
+
+fn generate_request(privkey: &PKey<Private>) -> Result<X509Req, ErrorStack> {
+    let mut req_builder = X509ReqBuilder::new()?;
+    req_builder.set_pubkey(&privkey)?;
+
+    let mut x509_name = X509NameBuilder::new()?;
+    x509_name.append_entry_by_text("C", "FR")?;
+    x509_name.append_entry_by_text("O", "Orbitale.io cert (dev)")?;
+    x509_name.append_entry_by_text("OU", "Orbitale localhost (dev)")?;
+    let x509_name = x509_name.build();
+    req_builder.set_subject_name(&x509_name)?;
+
+    req_builder.sign(&privkey, MessageDigest::sha256())?;
+    let req = req_builder.build();
+    Ok(req)
+}
+
+fn generate_ca_signed_cert(
+    ca_cert: &X509Ref,
+    ca_privkey: &PKeyRef<Private>,
+) -> Result<(X509, PKey<Private>), ErrorStack> {
+    let rsa = Rsa::generate(2048)?;
+    let privkey = PKey::from_rsa(rsa)?;
+
+    let req = generate_request(&privkey)?;
+
+    let mut cert_builder = X509::builder()?;
+    cert_builder.set_version(2)?;
+    let serial_number = {
+        let mut serial = BigNum::new()?;
+        serial.rand(159, MsbOption::MAYBE_ZERO, false)?;
+        serial.to_asn1_integer()?
+    };
+    cert_builder.set_serial_number(&serial_number)?;
+    cert_builder.set_subject_name(req.subject_name())?;
+    cert_builder.set_issuer_name(ca_cert.subject_name())?;
+    cert_builder.set_pubkey(&privkey)?;
+    let not_before = Asn1Time::days_from_now(0)?;
+    cert_builder.set_not_before(&not_before)?;
+    let not_after = Asn1Time::days_from_now(365)?;
+    cert_builder.set_not_after(&not_after)?;
+
+    cert_builder.append_extension(BasicConstraints::new().critical().build()?)?;
+
+    cert_builder.append_extension(
+        KeyUsage::new()
+            .critical()
+            .digital_signature()
+            .key_encipherment()
+            .build()?,
+    )?;
+    cert_builder.append_extension(
+        ExtendedKeyUsage::new().server_auth().build()?,
+    )?;
+
+    let subject_key_identifier =
+        SubjectKeyIdentifier::new().build(&cert_builder.x509v3_context(Some(ca_cert), None))?;
+    cert_builder.append_extension(subject_key_identifier)?;
+
+    let auth_key_identifier = AuthorityKeyIdentifier::new()
+        .keyid(false)
+        .issuer(false)
+        .build(&cert_builder.x509v3_context(Some(ca_cert), None))?;
+    cert_builder.append_extension(auth_key_identifier)?;
+
+    let subject_alt_name = SubjectAlternativeName::new()
+        .dns("localhost")
+        .ip("127.0.0.1")
+        .ip("::1")
+        .build(&cert_builder.x509v3_context(Some(ca_cert), None))?;
```

Comment:
Curiosity here: Do you think we could somehow "externalize" this part in order to allow users to create CA/certs for custom local domains in the future? Symfony CLI is doing it but only with the `.wip` domain, I'd personally like users to be able to add **any** domain (some people use `.docker` or `.local` for instance).

At some point, it could even use `runa` to open `/etc/hosts` / `c:\Windows\system32\drivers\etc\hosts` in order to add the domain and make it point to `127.0.0.1` without the user having to do it themselves (which would be kind of a revolution, since no dnsmasq config or anything, means a lot!)

Reply:
I have same idea. Add command to add domain into certificate.
The list of domain can be stored into `.rymfony` directory and used to update the `hosts` file (with specific command).

But :
* If a domain is already in `hosts` file, do not add.
* add comment into hosts file to prevent change this section? Like Flex add in `.env` file on install.

Preview:

```
### Rymfony Domain
127.0.0.1 my-domain.local
::1 my-domain.local
127.0.0.1 admin.my-domain.local
::1 admin.my-domain.local
### END Rymfony Domain
```



Add this feature is for an new PR?

Reply:
> If a domain is already in `hosts` file, do not add.

Seems obvious to me to avoid duplications 👍 


> add comment into hosts file to prevent change this section? Like Flex add in `.env` file on install.

I like the idea, Docker Desktop does it too, but I think it might over-complexify the process. If you already know how to do that with good performances, that may be good, but I think it might introduce a bit of overhead. Also, it is error-prone if anyone modifies the comment section (that's also one problem with Flex, by the way).

I think one line-comment could be good, like this:

```
127.0.0.1 my-domain.local # Added by Rymfony
::1 my-domain.local # Added by Rymfony
```

And for the "where do we put this", this could be on top of the file.
We could put it after all the comments lines that serve as disclaimer on Windows for example. It's easy to check (open the file, check the first characters of the line, and if it's `#` or a non-empty trimmed line, move to next line).

I'm also okay for a second PR :)